docs/src/main/asciidoc/bigquery.adoc

```diff
@@ -0,0 +1,91 @@
+This offers a convenient way to write pipelines where the goal is to load data into BigQuery.
+
+To begin using the `BigQueryFileMessageHandler`, start off by defining a `@ServiceActivator` bean for your application which returns the handler.
+
+[source,java]
+----
+@Bean
+@ServiceActivator(
+    inputChannel = "bigQueryInputChannel", outputChannel = "bigQueryOutputChannel")
+public MessageHandler messageSender(BigQuery bigQuery) {
+  return new BigQueryFileMessageHandler(bigQuery);
+}
+----
+
+There should be both an input and output channel configured for your `ServiceActivator`.
+
+* The input channel is responsible for accepting the https://docs.spring.io/spring-integration/docs/latest-ga/reference/html/#message[Messages] whose payloads contain the files you wish to load to your BigQuery table.
+* The output channel is where `BigQueryFileMessageHandler` will send all of the replies resulting from the load operations; each reply will yield a `Job` object which provides details about its corresponding load `Job`.
+
+==== Input Channel
+
+The `BigQueryFileMessageHandler` accepts messages whose payloads contain the data file being loaded.
+The accepted payload types include `byte[]`, `java.io.File`, or `java.io.InputStream`.
+
+Popular data file formats such as CSV or JSON are supported; see https://cloud.google.com/bigquery/docs/loading-data#supported_data_formats[the documentation] for a complete list.
+
+==== Output Channel
+
+On the provided output channel, the handler will return a https://googleapis.dev/java/google-cloud-clients/latest/com/google/cloud/bigquery/Job.html[Job] object or a `ListenableFuture<Job>` as a reply for each file that is loaded.
+
+Each `Job` object contains metadata and general statistics about the load job for the file.
+It will also indicate if any failures occurred.
+
+By default, the `BigQueryFileMessageHandler` is running in `async` mode, meaning that it will reply with `ListenableFuture<Job>` on the output channel.
+The returned `ListenableFuture` will complete when the load job is complete.
+
+If you call `bigQueryFileMessageHandler.setSync(true)`, it will run the job synchronously and return `Job` objects on the output channel (not wrapped in a Future object).
+The handler will block for each file to be loaded before moving on.
+
+=== Message Headers
+
+By default, the `BigQueryFileMessageHandler` is configured to read the message headers of each file message being sent through the input channel.
+This default can be modified by setting the dataset, table name, and format option expressions in `BigQueryFileMessageHandler`.
+
+The message headers determine where and how the file should be loaded to BigQuery.
+
+The following headers are available as constants in `BigQuerySpringMessageHeaders`:
+
+|===========================================================================
+| Message Header Name | Description
+| `DATASET_NAME` | Name of the BigQuery dataset that the file should be loaded to.
```

Comment:
It would be nice if we can also set the defaults for these through `spring.cloud.gcp.bigquery.*` properties so that it just becomes configuration, rather than code.